.github/workflows/ci.yml

```diff
@@ -78,14 +76,33 @@ jobs:
           shell: ${{ matrix.shell }}
           nim_ref: ${{ matrix.nim.ref }}
 
-      - name: Setup Go
-        uses: actions/setup-go@v5
-        with:
-          go-version: '~1.16.0' # That's the minimum Go version that works with arm.
 
       - name: Install p2pd
         run: |
-          V=1 bash scripts/build_p2pd.sh p2pdCache 124530a3
+          mkdir -p .ci-bin
+          if [ "${{ matrix.platform.os }}" = "windows" ]; then
+            curl -sLO "https://github.com/libp2p/go-libp2p-daemon/releases/download/v0.9.1/p2pd-windows-amd64.exe"
+            mv p2pd-windows-amd64.exe .ci-bin/p2pd.exe
+          elif [ "${{ matrix.platform.os }}" = "macos" ]; then
+            if [ "${{ matrix.platform.cpu }}" = "arm64" ]; then
+              curl -sLO "https://github.com/libp2p/go-libp2p-daemon/releases/download/v0.9.1/p2pd-darwin-arm64"
+              mv p2pd-darwin-arm64 .ci-bin/p2pd
+            else
+              curl -sLO "https://github.com/libp2p/go-libp2p-daemon/releases/download/v0.9.1/p2pd-darwin-amd64"
+              mv p2pd-darwin-amd64 .ci-bin/p2pd
+            fi
+            chmod +x .ci-bin/p2pd
+          else
+            if [ "${{ matrix.platform.cpu }}" = "i386" ]; then
+              echo "No official i386 build available for p2pd. Exiting..."
+              exit 1
+            else
+              curl -sLO "https://github.com/libp2p/go-libp2p-daemon/releases/download/v0.9.1/p2pd-linux-amd64"
+              mv p2pd-linux-amd64 .ci-bin/p2pd
+              chmod +x .ci-bin/p2pd
+            fi
+          fi
+          echo "${PWD}/.ci-bin" >> $GITHUB_PATH
```

Comment:
would it make sense to have this as some script under `.github` dir?

```
./install-p2p ${{ matrix.platform.os }} ${{ matrix.platform.cpu }}
```

or even better if it can be an action (in this repo `.github/actions/install_p2pd`) or even your action in your github so we just include it here.